project_stage_last_update_date/models/project_project.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2024 Tecnativa - Carolina Fernandez
+# License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+from datetime import datetime
+
+from odoo import fields, models
+
+
+class Project(models.Model):
+    _inherit = "project.project"
+
+    last_update_date_stage = fields.Datetime(copy=False, readonly=True, tracking=True)
+
+    def write(self, vals):
+        if "stage_id" in vals:
+            vals["last_update_date_stage"] = datetime.now()
```

Comment:
Open question do you want to store the technical datetime or the one of the user (different timezone, DST)?

Reply:
I see this one correct, as it's the server one (UTC), and will be displayed with tz according the user.